src/drawerbase.js

```diff
@@ -0,0 +1,291 @@
+ * @param {Object} options - Options for this Drawer.
+ * @param {OpenSeadragon.Viewer} options.viewer - The Viewer that owns this Drawer.
+ * @param {OpenSeadragon.Viewport} options.viewport - Reference to Viewer viewport.
+ * @param {Element} options.element - Parent element.
+ */
+
+$.DrawerBase = class DrawerBase{
+    constructor(options){
+        $.console.assert( options.viewer, "[Drawer] options.viewer is required" );
+
+        //backward compatibility for positional args while preferring more
+        //idiomatic javascript options object as the only argument
+        var args  = arguments;
+
+        if( !$.isPlainObject( options ) ){
+            options = {
+                source:     args[ 0 ], // Reference to Viewer tile source.
+                viewport:   args[ 1 ], // Reference to Viewer viewport.
+                element:    args[ 2 ]  // Parent element.
+            };
+        }
+
+        $.console.assert( options.viewport, "[Drawer] options.viewport is required" );
+        $.console.assert( options.element, "[Drawer] options.element is required" );
+
+        if ( options.source ) {
+            $.console.error( "[Drawer] options.source is no longer accepted; use TiledImage instead" );
+        }
+
+        this.viewer = options.viewer;
+        this.viewport = options.viewport;
+        this.debugGridColor = typeof options.debugGridColor === 'string' ? [options.debugGridColor] : options.debugGridColor || $.DEFAULT_SETTINGS.debugGridColor;
+        this.options = options.options || {};
+
+        /**
+         * The parent element of this Drawer instance, passed in when the Drawer was created.
+         * The parent of {@link OpenSeadragon.DrawerBase#canvas}.
+         * @member {Element} container
+         * @memberof OpenSeadragon.DrawerBase#
+         */
+        this.container  = $.getElement( options.element );
+
+        // We force our container to ltr because our drawing math doesn't work in rtl.
+        // This issue only affects our canvas renderer, but we do it always for consistency.
+        // Note that this means overlays you want to be rtl need to be explicitly set to rtl.
+        this.container.dir = 'ltr';
+
+        // the first time this.canvas is accessed, implementations will create it
+        this.canvas.style.width     = "100%";
+        this.canvas.style.height    = "100%";
+        this.canvas.style.position  = "absolute";
+        $.setElementOpacity( this.canvas, this.viewer.opacity, true ); // set
+
+        // Allow pointer events to pass through the canvas element so implicit
+        //   pointer capture works on touch devices
+        $.setElementPointerEventsNone( this.canvas );
+        $.setElementTouchActionNone( this.canvas );
+
+        // explicit left-align
+        this.container.style.textAlign = "left";
+        this.container.appendChild( this.canvas );
+
+        this._checkForAPIOverrides();
+    }
+
+    get canvas(){
+        if(!this._renderingTarget){
```

Comment:
This is neat, but why not just call `_createDrawingElement` in the constructor?

Reply:
Leftover from an earlier iteration of the design where this made more sense, I believe. Changed now. I'm leaving the public member `drawer` protected with the getter method since it should never be settable.

Reply:
Sounds good!